File: frontend/admin-web/api/dict.ts
/**
 * 系统数据字典 API 层
 */
import { service as http } from "@/utils/axios/service";

interface ApiResponse<T> {
  code: number;
  message: string;
  data: T;
}

// ─── 字典项 ──────────────────────────────────────────────────────────────────

export interface DictItem {
  id: string;
  dict_type: string;
  item_value: string;
  item_label: string;
  is_default: boolean;
  sort_order: number;
  description?: string;
}

export interface DictItemCreate {
  item_value: string;
  item_label: string;
  is_default?: boolean;
  sort_order?: number;
  description?: string;
}

export interface DictItemUpdate {
  item_value?: string;
  item_label?: string;
  is_default?: boolean;
  sort_order?: number;
  description?: string;
}

// ─── 字典类型 ────────────────────────────────────────────────────────────────

export interface SysDictSimple {
  id: string;
  dict_type: string;
  dict_name: string;
  description?: string;
  is_builtin: boolean;
  sort_order: number;
  item_count: number;
}

export interface DictCreate {
  dict_type: string;
  dict_name: string;
  description?: string;
  sort_order?: number;
}

export interface DictUpdate {
  dict_name?: string;
  description?: string;
  sort_order?: number;
}

// ─── API ─────────────────────────────────────────────────────────────────────

export const DictAPI = {
  // 字典类型
  list: () =>
    http.get<ApiResponse<SysDictSimple[]>, ApiResponse<SysDictSimple[]>>("/sys/dicts"),
  create: (data: DictCreate) =>
    http.post<ApiResponse<SysDictSimple>, ApiResponse<SysDictSimple>>("/sys/dicts", data),
  update: (dict_type: string, data: DictUpdate) =>
    http.put<ApiResponse<SysDictSimple>, ApiResponse<SysDictSimple>>(`/sys/dicts/${dict_type}`, data),
  remove: (dict_type: string) =>
    http.delete<ApiResponse<null>, ApiResponse<null>>(`/sys/dicts/${dict_type}`),

  // 字典项
  listItems: (dict_type: string) =>
    http.get<ApiResponse<DictItem[]>, ApiResponse<DictItem[]>>(`/sys/dicts/${dict_type}/items`),
  createItem: (dict_type: string, data: DictItemCreate) =>
    http.post<ApiResponse<DictItem>, ApiResponse<DictItem>>(`/sys/dicts/${dict_type}/items`, data),
  updateItem: (item_id: string, data: DictItemUpdate) =>
    http.put<ApiResponse<DictItem>, ApiResponse<DictItem>>(`/sys/dicts/items/${item_id}`, data),
  removeItem: (item_id: string) =>
    http.delete<ApiResponse<null>, ApiResponse<null>>(`/sys/dicts/items/${item_id}`),
};

/**
 * 便捷方法：将字典项转换为 Naive UI select options 格式
 */
export function dictToOptions(items: DictItem[]) {
  return items.map((i) => ({ label: i.item_label, value: i.item_value }));
}

File: backend/app/modules/iam/schemas/dict.py
"""数据字典 Pydantic schemas。"""
import uuid
from typing import Optional
from pydantic import BaseModel, Field


# ─── 字典项 ───────────────────────────────────────────────────────────────────

class DictItemCreate(BaseModel):
    item_value: str = Field(..., max_length=200, description="存储值")
    item_label: str = Field(..., max_length=200, description="显示标签")
    is_default: bool = False
    sort_order: int = 0
    description: Optional[str] = None


class DictItemUpdate(BaseModel):
    item_value: Optional[str] = Field(default=None, max_length=200)
    item_label: Optional[str] = Field(default=None, max_length=200)
    is_default: Optional[bool] = None
    sort_order: Optional[int] = None
    description: Optional[str] = None


class DictItemRead(BaseModel):
    id: uuid.UUID
    dict_type: str
    item_value: str
    item_label: str
    is_default: bool
    sort_order: int
    description: Optional[str] = None

    model_config = {"from_attributes": True}


# ─── 字典类型 ──────────────────────────────────────────────────────────────────

class DictCreate(BaseModel):
    dict_type: str = Field(..., max_length=64, description="字典代码，全局唯一")
    dict_name: str = Field(..., max_length=100, description="字典中文名")
    description: Optional[str] = None
    sort_order: int = 0


class DictUpdate(BaseModel):
    dict_name: Optional[str] = Field(default=None, max_length=100)
    description: Optional[str] = None
    sort_order: Optional[int] = None


class DictRead(BaseModel):
    id: uuid.UUID
    dict_type: str
    dict_name: str
    description: Optional[str] = None
    is_builtin: bool
    sort_order: int
    items: list[DictItemRead] = []

    model_config = {"from_attributes": True}


class DictReadSimple(BaseModel):
    """不含 items 的轻量版，用于列表接口。"""
    id: uuid.UUID
    dict_type: str
    dict_name: str
    description: Optional[str] = None
    is_builtin: bool
    sort_order: int
    item_count: int = 0

    model_config = {"from_attributes": True}
